crates/compiler/builtins/roc/List.roc

```diff
@@ -528,6 +529,24 @@ dropIf : List a, (a -> Bool) -> List a
 dropIf = \list, predicate ->
     List.keepIf list (\e -> Bool.not (predicate e))
 
+## Run the given function on each element of a list, and return the
+## number of elements for which the function returned `Bool.true`.
+countIf : List a, (a -> Bool) -> Nat
+countIf = \list, predicate ->
+    length = List.len list
+
+    countIfHelp list predicate 0 0 length
```

Comment:
Instead of rolling your own function here, can you use `List.walk`?

The state would be the count and you would run the function on each element.